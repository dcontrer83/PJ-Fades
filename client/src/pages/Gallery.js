import React, {useState} from 'react';
import {useMediaQuery} from 'react-responsive'
import {motion} from 'framer-motion'
// either import image from a folder
// import imageTest1 from '../img/imgTest1.png';
import images from '../images'
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

const Gallery = () => {

    const [currImage, setImage] = useState(0)
    const handleClick = (num) => {
        const index = num;
        setImage(index);
    }

    const isLaptop = useMediaQuery({query: '(min-width: 1024px)',})

    const style = {
        offset: {
            height: "75px"
        },
        main: {
            height: "1000px"
        },
        mainOff: {
            height: "1000px"
        },
        content1: {
            height: "90%",
            width: "100%"
        },
        content2: {
            height: "50%",
            width: "100%"
        },
        mainPic: {
            height: "100%",
            width: "auto",
            objectFit: "contain"
        },
        onePic: {
            height: "auto",
            width: "90%"
        },
        pics: {
            height: "150px",
            width: "150px",
            objectFit: "cover",
            borderRadius: "15px",
            cursor: "pointer",
            borderStyle: "solid"
        },
        box: {
            height: "95%",
            width: "100%"
        }
    }
    return (
        <main>
            <div data-testid="div-1-element" style={style.offset}></div>
            <div data-testid="div-2-element" style={style.main}>
                <h1 data-testid="h1-element" className='text-center mt-5'>Gallery</h1>
                <div data-testid="div-3-element" className='d-flex flex-column flex-lg-row align-items-center' style={style.box}>
                    <div data-testid="div-4-element" className='bg-light d-flex justify-content-center align-items-center' style={isLaptop ? style.content1 : style.content2}>
                        {currImage === 0 || currImage === 5 ? 
                        <img data-testid="main-img-1-element" style={style.onePic} src={images[currImage]} alt="expanded"></img> : 
                        <img data-testid="main-img-2-element" style={style.mainPic} src={images[currImage]} alt="expanded"></img>
                        }
                    </div>
                    <div data-testid="div-5-element" className='d-flex flex-row bg-light' style={isLaptop ? style.content1 : style.content2}>
                        <Row data-testid="row-1-element" className='text-center mt-3' sm={1}  >
                            <Col><motion.img data-testid="img-0-element" whileHover={{scale:1.1}} src={images[0]} style={style.pics} onClick={() => handleClick(0)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-1-element" whileHover={{scale:1.1}} src={images[1]} style={style.pics} onClick={() => handleClick(1)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-2-element" whileHover={{scale:1.1}} src={images[2]} style={style.pics} onClick={() => handleClick(2)} alt="hero"></motion.img></Col>
                        </Row>
                        <Row data-testid="row-2-element" className='text-center mt-3' sm={1} >
                            <Col><motion.img data-testid="img-3-element" whileHover={{scale:1.1}} src={images[3]} style={style.pics} onClick={() => handleClick(3)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-4-element" whileHover={{scale:1.1}} src={images[4]} style={style.pics} onClick={() => handleClick(4)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-5-element" whileHover={{scale:1.1}} src={images[5]} style={style.pics} onClick={() => handleClick(5)} alt="hero"></motion.img></Col>
                        </Row>
                        <Row data-testid="row-3-element" className='text-center mt-3' sm={1}>
                            <Col><motion.img data-testid="img-6-element" whileHover={{scale:1.1}} src={images[6]} style={style.pics} onClick={() => handleClick(6)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-7-element" whileHover={{scale:1.1}} src={images[7]} style={style.pics} onClick={() => handleClick(7)} alt="hero"></motion.img></Col>
                            <Col><motion.img data-testid="img-8-element" whileHover={{scale:1.1}} src={images[8]} style={style.pics} onClick={() => handleClick(8)} alt="hero"></motion.img></Col>
                        </Row>
                    </div>
                </div>
            </div>
        </main>
    )
}

export default Gallery;