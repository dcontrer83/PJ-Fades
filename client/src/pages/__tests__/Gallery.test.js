import React from 'react';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Gallery from '../Gallery';
import '@testing-library/jest-dom';
import { render, screen, fireEvent, waitFor, getByTestId } from '@testing-library/react';

globalThis.IS_REACT_ACT_ENVIRONMENT = true;

let root;
let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    act(() => {
        root = createRoot(container);
    })
});

afterEach(() => {
    act(() => {
        root.unmount();
    })
    container.remove();
    container = null;
});

let div1Element;
let div2Element;
let div3Element;
let div4Element;
let div5Element;

function getAllDivElements() {
    render(<Gallery />);
    div1Element = screen.getByTestId('div-1-element');
    div2Element = screen.getByTestId('div-2-element');
    div3Element = screen.getByTestId('div-3-element');
    div4Element = screen.getByTestId('div-4-element');
    div5Element = screen.getByTestId('div-5-element');
}

describe('Div elements', () => {
    it('should be in the document', () => {
        getAllDivElements();
        expect(div1Element).toBeInTheDocument();
        expect(div2Element).toBeInTheDocument();
        expect(div3Element).toBeInTheDocument();
        expect(div4Element).toBeInTheDocument();
        expect(div5Element).toBeInTheDocument();
    });
    it('should have specific style', () => {
        getAllDivElements();
        expect(div1Element).toHaveStyle('height: 75px');
        expect(div2Element).toHaveStyle('height: 1000px');
        expect(div3Element).toHaveStyle('height: 95%; width: 100%');
        expect(div4Element).toHaveStyle('height: 50%; width: 100%');
        expect(div4Element).not.toHaveStyle('height: 90%');
        expect(div5Element).toHaveStyle('height: 50%; width: 100%');
        expect(div5Element).not.toHaveStyle('height: 90%');
    });
    it('should have specific className', () => {
        getAllDivElements();
        expect(div3Element).toHaveClass('d-flex flex-column flex-lg-row align-items-center');
        expect(div4Element).toHaveClass('bg-light d-flex justify-content-center align-items-center');
        expect(div5Element).toHaveClass('d-flex flex-row bg-light');
    });
});

let h1Element;

function getH1Element() {
    render(<Gallery />);
    h1Element = screen.getByTestId('h1-element');
}

describe('H1 element', () => {
    it('should be in the document', () => {
        getH1Element();
        expect(h1Element).toBeInTheDocument();
    });
    it('should have specific className', () => {
        getH1Element();
        expect(h1Element).toHaveClass('text-center mt-5');
    });
    it('should have specific text content', () => {
        getH1Element();
        expect(h1Element).toHaveTextContent('Gallery');
    });
});

let mainImg1Element;
let mainImg2Element;
let img0Element;
let img1Element;
let img2Element;
let img3Element;
let img4Element;
let img5Element;
let img6Element;
let img7Element;
let img8Element;

function getAllImgElements() {
    render(<Gallery />);
    mainImg1Element = screen.getByTestId('main-img-1-element');
    img0Element = screen.getByTestId('img-0-element');
    img1Element = screen.getByTestId('img-1-element');
    img2Element = screen.getByTestId('img-2-element');
    img3Element = screen.getByTestId('img-3-element');
    img4Element = screen.getByTestId('img-4-element');
    img5Element = screen.getByTestId('img-5-element');
    img6Element = screen.getByTestId('img-6-element');
    img7Element = screen.getByTestId('img-7-element');
    img8Element = screen.getByTestId('img-8-element');
}

describe('Img elements', () => {
    it('main img 1 should be in the document', () => {
        getAllImgElements();
        expect(mainImg1Element).toBeInTheDocument();
    });
    it('main img 2 should be in the document',  () => {
        getAllImgElements();
        fireEvent.click(img1Element);
        mainImg2Element = screen.getByTestId('main-img-2-element');
        expect(mainImg2Element).toBeInTheDocument();
    });
    it('all img elements should be in the document', () => {
        getAllImgElements();
        expect(img0Element).toBeInTheDocument();
        expect(img1Element).toBeInTheDocument();
        expect(img2Element).toBeInTheDocument();
        expect(img3Element).toBeInTheDocument();
        expect(img4Element).toBeInTheDocument();
        expect(img5Element).toBeInTheDocument();
        expect(img6Element).toBeInTheDocument();
        expect(img7Element).toBeInTheDocument();
        expect(img8Element).toBeInTheDocument();
    });
    it('should have specific style', () => {
        getAllImgElements();
        expect(mainImg1Element).toHaveStyle('height: auto; width: 90%');
        expect(img0Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img1Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img2Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img3Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img4Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img5Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img6Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img7Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        expect(img8Element).toHaveStyle('height: 150px; width: 150px; objectFit: cover; borderRadius: 15px; cursor: pointer; borderStyle: solid;');
        fireEvent.click(img1Element);
        mainImg2Element = screen.getByTestId('main-img-2-element');
        expect(mainImg2Element).toHaveStyle('height: 100%; width: auto; objectFit: contain');
    });
    it('should have proper src attributes', () => {
        getAllImgElements();
        expect(mainImg1Element).toHaveAttribute('src', 'DSC01193.jpg');
        expect(img0Element).toHaveAttribute('src', 'DSC01193.jpg');
        expect(img1Element).toHaveAttribute('src', 'Hero-Gallery.PNG');
        expect(img2Element).toHaveAttribute('src', 'IMG_0525.jpg');
        expect(img3Element).toHaveAttribute('src', 'IMG_3489.jpg');
        expect(img4Element).toHaveAttribute('src', 'IMG_3735.jpg');
        expect(img5Element).toHaveAttribute('src', 'IMG_4210.jpg');
        expect(img6Element).toHaveAttribute('src', 'IMG_9194.jpg');
        expect(img7Element).toHaveAttribute('src', 'SampleStyle1.PNG');
        expect(img8Element).toHaveAttribute('src', 'SampleStyle2.PNG');
        fireEvent.click(img1Element);
        mainImg2Element = screen.getByTestId('main-img-2-element');
        expect(mainImg2Element).toHaveAttribute('src', 'Hero-Gallery.PNG');
    });
    it('should have proper alt attributes', () => {
        getAllImgElements();
        expect(mainImg1Element).toHaveAttribute('alt', 'expanded');
        expect(img0Element).toHaveAttribute('alt', 'hero');
        expect(img1Element).toHaveAttribute('alt', 'hero');
        expect(img2Element).toHaveAttribute('alt', 'hero');
        expect(img3Element).toHaveAttribute('alt', 'hero');
        expect(img4Element).toHaveAttribute('alt', 'hero');
        expect(img5Element).toHaveAttribute('alt', 'hero');
        expect(img6Element).toHaveAttribute('alt', 'hero');
        expect(img7Element).toHaveAttribute('alt', 'hero');
        expect(img8Element).toHaveAttribute('alt', 'hero');
        fireEvent.click(img1Element);
        mainImg2Element = screen.getByTestId('main-img-2-element');
        expect(mainImg2Element).toHaveAttribute('alt', 'expanded');
    });
    it('should change scale when mouse hover over', async () => {
        getAllImgElements();
        fireEvent.mouseEnter(img0Element);
        await waitFor(() => expect(img0Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img1Element);
        await waitFor(() => expect(img1Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img2Element);
        await waitFor(() => expect(img2Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img3Element);
        await waitFor(() => expect(img3Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img4Element);
        await waitFor(() => expect(img4Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img5Element);
        await waitFor(() => expect(img5Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img6Element);
        await waitFor(() => expect(img6Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img7Element);
        await waitFor(() => expect(img7Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
        fireEvent.mouseEnter(img8Element);
        await waitFor(() => expect(img8Element).toHaveStyle('transform: scale(1.1) translateZ(0)'), { timeout: 3000 });
    });
});

let row1Element;
let row2Element;
let row3Element;

function getAllRowElements() {
    render(<Gallery />);
    row1Element = screen.getByTestId('row-1-element');
    row2Element = screen.getByTestId('row-2-element');
    row3Element = screen.getByTestId('row-3-element');
}

describe('Row elements', () => {
    it('should be in the document', () => {
        getAllRowElements();
        expect(row1Element).toBeInTheDocument();
        expect(row2Element).toBeInTheDocument();
        expect(row3Element).toBeInTheDocument();
    });
    it('should have specific className', () => {
        getAllRowElements();
        expect(row1Element).toHaveClass('text-center mt-3 row-cols-sm-1');
        expect(row2Element).toHaveClass('text-center mt-3 row-cols-sm-1');
        expect(row3Element).toHaveClass('text-center mt-3 row-cols-sm-1');
    });
});

describe('Gallery', () => {
    it('should match snapshot', () => {
        act(() => {
            root.render(<Gallery />);
        });
        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});
