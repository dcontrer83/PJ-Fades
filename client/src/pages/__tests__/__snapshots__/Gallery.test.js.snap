// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should match snapshot 1`] = `
"<main>
  <div data-testid=\\"div-1-element\\" style=\\"height: 75px;\\"></div>
  <div data-testid=\\"div-2-element\\" style=\\"height: 1000px;\\">
    <h1 data-testid=\\"h1-element\\" class=\\"text-center mt-5\\">Gallery</h1>
    <div data-testid=\\"div-3-element\\" class=\\"d-flex flex-column flex-lg-row align-items-center\\" style=\\"height: 95%; width: 100%;\\">
      <div data-testid=\\"div-4-element\\" class=\\"bg-light d-flex justify-content-center align-items-center\\" style=\\"height: 50%; width: 100%;\\"><img data-testid=\\"main-img-1-element\\" style=\\"height: auto; width: 90%;\\" src=\\"DSC01193.jpg\\" alt=\\"expanded\\"></div>
      <div data-testid=\\"div-5-element\\" class=\\"d-flex flex-row bg-light\\" style=\\"height: 50%; width: 100%;\\">
        <div data-testid=\\"row-1-element\\" class=\\"text-center mt-3 row row-cols-sm-1\\">
          <div class=\\"col\\"><img data-testid=\\"img-0-element\\" src=\\"DSC01193.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-1-element\\" src=\\"Hero-Gallery.PNG\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-2-element\\" src=\\"IMG_0525.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
        </div>
        <div data-testid=\\"row-2-element\\" class=\\"text-center mt-3 row row-cols-sm-1\\">
          <div class=\\"col\\"><img data-testid=\\"img-3-element\\" src=\\"IMG_3489.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-4-element\\" src=\\"IMG_3735.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-5-element\\" src=\\"IMG_4210.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
        </div>
        <div data-testid=\\"row-3-element\\" class=\\"text-center mt-3 row row-cols-sm-1\\">
          <div class=\\"col\\"><img data-testid=\\"img-6-element\\" src=\\"IMG_9194.jpg\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-7-element\\" src=\\"SampleStyle1.PNG\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
          <div class=\\"col\\"><img data-testid=\\"img-8-element\\" src=\\"SampleStyle2.PNG\\" style=\\"height: 150px; width: 150px; object-fit: cover; border-radius: 15px; cursor: pointer; border-style: solid;\\" alt=\\"hero\\"></div>
        </div>
      </div>
    </div>
  </div>
</main>"
`;
